### PR TITLE
Fix e2e auth logic + add namespace a11y playwright tests

### PR DESCRIPTION
**Pull request by vyncent-t**
### Description

This PR improves accessibility testing coverage and reliability of Playwright E2E tests in the project.

Note: this also fixes some e2e test failures for my other branches, once this is merged in I will drop the duplicates in the other branches.

- https://github.com/kubernetes-sigs/headlamp/pull/3124
- https://github.com/kubernetes-sigs/headlamp/pull/3123

#### Accessibility Enhancements

- Adds `@axe-core/playwright` integration to the Namespaces Playwright test to perform automated **A11Y checks**.
- Introduces accessibility assertions to verify that no critical violations occur before and after creating a namespace.
- Fixes an issue detected by these tests: the **"Create" button lacked an accessible label**. This was resolved by adding an `aria-label` to ensure compliance with screen reader standards.

#### Navigation & Auth Logic Improvements

- Updates the **navigation and authentication logic** used in our Playwright E2E test setup.
(Previously, navigation during multi-cluster testing was **unreliable and prone to hanging or timing out** due to race conditions and cluster token loading delays.)
- The logic has been refactored to **work consistently in both CI and local testing environments**, with better handling of cluster contexts and token usage.

### Notes

- No issue was filed for this work — these improvements were identified during test development.
- More accessibility and stability improvements are planned in follow-up PRs to expand coverage across additional areas of the UI.

### CODE REVIEW

File: frontend/src/components/common/CreateResourceButton.tsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ export function CreateResourceButton(props: CreateResourceButtonProps) {
         errorMessage={errorMessage}
         onEditorChanged={() => setErrorMessage('')}
         title={t('translation|Create {{ name }}', { name })}
+        aria-label={t('translation|Create resource button')}
```
Comment (illume) [inline frontend/src/components/common/CreateResourceButton.tsx:39]: - Probably should be fixed inside EditorDialog to use the title.
- Using the same name as the title would be better, because it's more correct and we already have translations for it.

**Review by illume — CHANGES_REQUESTED** [review]
Nice work.

I just requested one change to the aria-label to use the existing title translation.

**Review by illume — APPROVED** [review]
This is great. Thank you!
